### DISCUSSION TITLE
Dynamic field resolvers

### DISCUSSION THREAD

**Discussion by FrederikJorgensen** [MichalLytek/type-graphql#1177] [Q&A]
Is it possible to somehow generate "dynamic" field resolvers? Such that from this object type:
```
@ObjectType()
class User {
  name: string
  photos?: Photo[]
  posts?: Post[]
}
```
It could generate the field resolvers on this resolver:
```
@Resolver(User)
class UserResolver {
  @FieldResolver(() => [Photo])
  public async photos() {}

  @FieldResolver(() => [Post])
  public async posts() {}
}
```

**Comment by MichalLytek** [comment]
Sorry, once again: what is your question? 🤔

**Reply by FrederikJorgensen** [reply]
I have a bunch of field resolvers that share some logic. I don't want to copy & paste `@FieldResolver()` all the time. 

Basically, I want to have an entity where I can just add properties to the `@ObjectType` and those would be dynamically added to the `@Resolver`.
```
@ObjectType()
class User {
  photos?: Photo[]
  ....100 more properties
}
```
```
@Resolver(User)
class UserResolver {
  @FieldResolver(() => [Photos])
  public async photos() {}
  ....100 more field resolvers
}

**Reply by MichalLytek** [reply]
Without know what is your shared logic, I can't suggest any solution.

**Reply by FrederikJorgensen** [reply]
It's just getting some data from TypeORM.
```
function sharedLogic() {
  const relationMapper = new RelationMapper(getConnection());
  const relations = relationMapper.buildRelationListForQuery(entity, info);
  options.filter = {};
  const [items, count] = await service.findAndCount(options, relations);
  return { items, count };
}
```
Then I expect a result like this back:
```
{
  "user": {
    "photos": {
      "items": [],
      "count": 0
    }
  }
}
```

**Reply by MichalLytek** [reply]
Have you tried writing generic resolvers?
